### PR TITLE
Add streamline "Create new note" command

### LINKED ISSUE #562: [Templates v2] Create a `Foam: Create New Note` command
As described in #534, create a `Foam: Create New Note` command which is a streamlined version of the `Foam: Create New Note From Template` command.

After this first implementation, it will reduce the number of steps required to create a new note (relative to `Foam: Create New Note From Template`) by defaulting to use the template found at `.foam/templates/new-note.md` (or an implicit template if `.foam/templates/new-note.md` does not exist).

### Future steps

As described in #534, as the rest of `Templates v2` gets implemented, `Foam: Create New Note` will become more and more streamlined, earning its position as the default way to create new notes in Foam.

### PR DESCRIPTION
**Pull request by movermeyer**
Closes #562

This PR creates a new `Foam: Create New Note` command that serves as a streamlined version of `Foam: Create New Note From Template`, as described in #534.

### Open questions

I have used the term "Default template" to refer to this special `new-note.md` template.
I'm not 100% satisfied with this name, but didn't come up with a better name. Perhaps you can?

### Next steps

As we implement the rest of the steps of #534, this command will become more and more streamlined, in many cases, it won't have to ask for any user input at all.

### CODE REVIEW

**Review by ingalless — COMMENTED** [review]
Like this! Do you think it's possible to add a test case to the test suite for this?

File: docs/features/creating-new-notes.md [hunk 1]
```diff
@@ -2,11 +2,13 @@
 
 - Write out a new `[[wiki-link]]` and `Cmd` + `Click` to create a new file and enter it.
   - For keyboard navigation, use the 'Follow Definition' key `F12` (or [remap key binding](https://code.visualstudio.com/docs/getstarted/keybindings) to something more ergonomic)
-- `Cmd` + `Shift` + `P` (`Ctrl` + `Shift` + `P` for Windows), execute `New Note` from [VS Code Markdown Notes](https://marketplace.visualstudio.com/items?itemName=kortina.vscode-markdown-notes) and enter a **Title Case Name** to create `title-case-name.md`
+- `Cmd` + `Shift` + `P` (`Ctrl` + `Shift` + `P` for Windows), execute `Foam: Create New Note` and enter a **Title Case Name** to create `Title Case Name.md`
```
Comment (revett) [inline docs/features/creating-new-notes.md:5]: Do we want to move away from using `kebab-case` for filenames?

Comment (riccardoferretti) [reply docs/features/creating-new-notes.md:5]: yes, we are following the Obsidian convention, not the Markdown Notes one

**Comment by riccardoferretti** [comment]
at a high level this looks good. Can you rebase from master, as I just merged a pretty big refactoring of the core model. Shouldn't have impacted you too much, but just to be sure..

**Comment by riccardoferretti** [comment]
Can I also ask you to play with Foam to see whether the refactoring introduced any regression? I have tested it extensively, but might be missing something because of the way I use it and the platform I am on

**Review by riccardoferretti — APPROVED** [review]
LGTM!

**Pull request merged**